### DISCUSSION TITLE
Support crates for onboarding/tutorial popups?

### DISCUSSION THREAD

**Discussion by stephanemagnenat** [emilk/egui#2732] [Q&A]
Hello,

Does anyone know whether there exists some support crate that provides the feature of simple explanation popup? To target use is in a strategy game, using bevy and egui, when explaining the UI to the user. Ideally, the rest of the UI will be disabled for click, and possibly part of it shaded. That will allow to nicely introduce the UI of the game to a user being onboarded as part of a tutorial phase.

I know there exists support for notifications, but I did not find anything for onboarding/tutorial.

Thank you!

**Comment by dscso** [comment]
I just implemented something like that

```rust
/// A popup window with a title and a close button.
/// The popup is centered on the screen, and not movable/resizable.
/// add_contents is a closure that adds the contents of the popup.
pub fn popup(
    ctx: &egui::Context,
    title: &str,
    open: &mut bool,
    add_contents: impl FnOnce(&mut Ui),
) {
    ctx.input(|e| {
        if e.key_pressed(egui::Key::Escape) {
            *open = false;
        }
    });
    Window::new(title)
        .resizable(false)
        .collapsible(false)
        .movable(false)
        .open(open)
        //.fixed_size(egui::vec2(200.0, 300.0))
        .anchor(Align2::CENTER_CENTER, egui::vec2(0.0, 0.0))
        .show(ctx, add_contents);
}
```

then I used the `fn update` 
the function `ui.set_enabled(!self.login_panel.open);` pretty simple and effective
